src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs

```diff
@@ -104,15 +104,27 @@ private string ContextBuilder
 				var e = base.VisitMemberAccessExpression(node);
 				var isValidParent = !Helpers.IsInsideMethod(node).result && !Helpers.IsInsideMemberAccessExpression(node).result;
 				var isParentMemberStatic = node.Expression is MemberAccessExpressionSyntax m && _isStaticMember(m.ToFullString());
+				var isPathLessCast = Helpers.IsPathLessCast(node);
 
 				if (e!= null && isValidParent && !_isStaticMember(node.Expression.ToFullString()) && !isParentMemberStatic)
 				{
-					var expression = e.ToFullString();
-					var contextBuilder = _isStaticMember(expression) ? "" : ContextBuilder;
-					var output = ParseCompilationUnit($"class __Temp {{ private Func<object> __prop => {contextBuilder}{expression}; }}");
+					if (isPathLessCast.result)
+					{
+						var expression = e.ToFullString();
+						var output = ParseCompilationUnit($"class __Temp {{ private Func<object> __prop => ({isPathLessCast.expression?.Expression}){_contextName}; }}");
 
-					var o2 = output.DescendantNodes().OfType<ArrowExpressionClauseSyntax>().First().Expression;
-					return o2;
+						var newSyntax = output.DescendantNodes().OfType<ArrowExpressionClauseSyntax>().First().Expression;
+						return newSyntax;
+					}
+					else
+					{
+						var expression = e.ToFullString();
+						var contextBuilder = _isStaticMember(expression) ? "" : ContextBuilder;
+						var output = ParseCompilationUnit($"class __Temp {{ private Func<object> __prop => {contextBuilder}{expression}; }}");
+
+						var newSyntax = output.DescendantNodes().OfType<ArrowExpressionClauseSyntax>().First().Expression;
+						return newSyntax;
```

Comment:
Looks like this is more work than needed? Isn't `return SyntaxFactory.ParseExpression($"{contextBuilder}{expression};");` what we need?
We are creating a whole compilation unit just to get the expression syntax from it. Any thing I'm missing?

Reply:
It's only a `CompilationUnit` syntax, and the idea behind it is to have a proper resolution context so that the nodes returned are from a method body, not just a top-level statement that may not make sense alone.

Reply:
Still, this part is duplicated many times, it's time for a refactor ;)

Reply:
Is there any known case where ParseExpression will not produce the expected result?

I did a quick benchmark and the ParseExpression approach appears to be faster:

![image](https://user-images.githubusercontent.com/31348972/192887951-248865b7-1eff-444f-8298-09c121f47763.png)

Not sure if the difference is going to be noticeable in our case, but something to consider if it maintains the code correctness.

Reply:
It's an interesting benchmark. It's only used in smaller use cases when x:Bind expressions are used, but it can make it faster anyways, I'll check in the other PR. Thanks!